File: Python/Problem_005.py
# Author: Christian Seely

import functools

def gcd(a,b):
    while b:
        r = a%b
        a, b = b, r
    return a

def lcm(a,b):
    return a/gcd(a,b)*b

def problemFive():
    return int(functools.reduce(lcm, range(1,21)))


if __name__ == '__main__':
    print (problemFive())

File: C++/Problem_005.cpp
#include "stdafx.h"
#include <iostream>
#include <algorithm>
#include <vector>
#include <numeric>
#include <math.h>
#include <execution>

/*Author: Christian Seely*/

int problemFive()
{
  std::vector<int> range(20);
  std::iota(range.begin(), range.end(), 1);
  // Execution method: parallel, start, end, initial acc, custom reduction lambda. 
  // Note, warnings for experimental method usage may need to be suppressed.  (SDL checks for visual studio).
  return std::reduce(std::execution::par, range.begin(), range.end(), 1, [](int a, int b) {return a / std::gcd(a, b)*b;});
}


int main()
{
  std::cout << problemFive() << std::endl;
  return 0;
}

